pkg/fioapp/layers.go

```diff
@@ -83,16 +79,49 @@ func GetAppLayersFromMap(ctx context.Context, svcImages map[string]string) (map[
 			return nil, err
 		}
 
-		manifestList, err := GetManifestList(ctx, imageManifestSvc, imageRef.(reference.Canonical).Digest())
+		indexManifest, err := imageManifestSvc.Get(ctx, imageRef.(reference.Canonical).Digest())
 		if err != nil {
 			return nil, err
 		}
 
-		for _, manifest := range manifestList.Manifests {
-			if _, ok := archToManifestList[manifest.Platform.Architecture]; !ok {
-				archToManifestList[manifest.Platform.Architecture] = make(map[distribution.ManifestService]digest.Digest)
+		switch im := indexManifest.(type) {
+		case *manifestlist.DeserializedManifestList:
+			{
+				manifestList := im
+				for _, manifest := range manifestList.Manifests {
+					if _, ok := archToManifestList[manifest.Platform.Architecture]; !ok {
+						archToManifestList[manifest.Platform.Architecture] = make(map[distribution.ManifestService]digest.Digest)
+					}
+					archToManifestList[manifest.Platform.Architecture][imageManifestSvc] = manifest.Digest
+				}
+				break
```

Comment:
you don't need `break`s in golang.

Reply:
That's because I was writing lots of C++ code before :).